### DISCUSSION TITLE
BrowserRouter does not work with electron but HashRouter is not recommended

### DISCUSSION THREAD

**Discussion by ozgeneral** [remix-run/react-router#10724] [Q&A]
I am trying to port a webpage to electron which uses BrowserRouter

this fails with "[react-router] Location "/path/to/index.html" did not match any routes"

which seems to be a common issue:
https://stackoverflow.com/questions/36505404/how-to-use-react-router-with-electron

workaround people have came up with seems to be to use HashRouter instead. However [documentation](https://reactrouter.com/en/main/router-components/hash-router) says "We strongly recommend you do not use HashRouter unless you absolutely have to" without any further context. 

So I would like to get information about: 
 - what is the recommended way to use react router with electron?
 - what is the drawback of using HashRouter?
 - is it intended that BrowserRouter fails with electron?

**Comment by abhijith94** [comment]
+1

**Comment by GabsxBatata** [comment]
Bump

**Comment by R-Bower** [comment]
Is it possible to use HashRouter in framework mode? I want to use the file-based routing to make things consistent between my web apps and desktop apps.  But I can't seem to get HashRouter working in framework mode with Electron.

**Comment by kelsin** [comment]
I use HashRouter in my electron app on v7 and have had no issues. I consider that use case "unless you absolutely have to" since we don't have control and the users don't interact with the location directly.

**Comment by Wiktor102** [comment]
I encountered the same issue. I've decided to use `MemoryRouter` instead, and it works. There isn't much [documentation](https://reactrouter.com/api/declarative-routers/MemoryRouter) about it, so I don't know if it's better or worse than HashRouter.

__I also possibly see another solution, but it probably isn't worth the overhead:__
As mentioned in the [quoted SO post](https://stackoverflow.com/questions/36505404/how-to-use-react-router-with-electron), the `BrowserRouter` only doesn't work if loading the page with `mainWindow.loadURL('file://path-on-disk.html');`. It works correctly if we pass an url. So we could setup a simple web server inside the main process and use that.
